splunk-otel-android/src/main/java/com/splunk/rum/SplunkSpanDataModifier.java

```diff
@@ -74,6 +81,15 @@ private SpanData modify(SpanData original) {
         List<EventData> modifiedEvents = new ArrayList<>(original.getEvents().size());
         AttributesBuilder modifiedAttributes = original.getAttributes().toBuilder();
 
+        SpanContext spanContext;
+        if (reactNativeEnabled) {
+            spanContext = extractReactNativeIdsIfPresent(original);
+            modifiedAttributes.remove(REACT_NATIVE_TRACE_ID_KEY);
+            modifiedAttributes.remove(REACT_NATIVE_SPAN_ID_KEY);
```

Comment:
We can change this later, but I think it's unfortunate that we have to deal with the react stuff here. I wish this was done somehow via composition -- with another exporter or "data modifier" that wrapped this (or that this wrapped) only got wired up when RN support was enabled.

Reply:
I've done it this way to avoid creating another `SpanData` copy (cause this is Android, and we probably should be careful and avoid too many allocations on the hot path).
But, now that I think of it, I guess we could split this class into several parts (implementing an internal interface) and pass a mutable `SplunkSpanBuilder` between them - I'll explore this in another PR.

Reply:
something something premature optimization.... 🙃